README.md
# x README

This is the README for your extension "x". After writing up a brief description, we recommend including the following sections.

## Features

Describe specific features of your extension including screenshots of your extension in action. Image paths are relative to this README file.

For example if there is an image subfolder under your extension project workspace:

\!\[feature X\]\(images/feature-x.png\)

> Tip: Many popular extensions utilize animations. This is an excellent way to show off your extension! We recommend short, focused animations that are easy to follow.

## Requirements

If you have any requirements or dependencies, add a section describing those and how to install and configure them.

## Extension Settings

Include if your extension adds any VS Code settings through the `contributes.configuration` extension point.

For example:

This extension contributes the following settings:

* `myExtension.enable`: enable/disable this extension
* `myExtension.thing`: set to `blah` to do something

## Known Issues

Calling out known issues can help limit users opening duplicate issues against your extension.

## Release Notes

Users appreciate release notes as you update your extension.

### 1.0.0

Initial release of ...

### 1.0.1

Fixed issue #.

### 1.1.0

Added features X, Y, and Z.

-----------------------------------------------------------------------------------------------------------

## Working with Markdown

**Note:** You can author your README using Visual Studio Code.  Here are some useful editor keyboard shortcuts:

* Split the editor (`Cmd+\` on OSX or `Ctrl+\` on Windows and Linux)
* Toggle preview (`Shift+CMD+V` on OSX or `Shift+Ctrl+V` on Windows and Linux)
* Press `Ctrl+Space` (Windows, Linux) or `Cmd+Space` (OSX) to see a list of Markdown snippets

### For more information

* [Visual Studio Code's Markdown Support](http://code.visualstudio.com/docs/languages/markdown)
* [Markdown Syntax Reference](https://help.github.com/articles/markdown-basics/)

**Enjoy!**

CHANGELOG.md
# Change Log
All notable changes to the "x" extension will be documented in this file.

Check [Keep a Changelog](http://keepachangelog.com/) for recommendations on how to structure this file.

## [Unreleased]
- Initial release

out/src/extension.js
'use strict';
Object.defineProperty(exports, "__esModule", { value: true });
// The module 'vscode' contains the VS Code extensibility API
// Import the module and reference it with the alias vscode in your code below
const vscode_1 = require("vscode");
// this method is called when your extension is activated
// your extension is activated the very first time the command is executed
function activate(context) {
    // Use the console to output diagnostic information (console.log) and errors (console.error)
    // This line of code will only be executed once when your extension is activated
    console.log('Congratulations, your extension "x" is now active!');
    // Get settings
    //let settings = workspace.getConfiguration().get("tomatoTimer");
    let shortToLongTime = 3;
    let timerStart = vscode_1.commands.registerCommand('pomodoro.start', () => {
        vscode_1.window
            .showQuickPick(['10 minutes', '20 minutes', '25 minutes', '45 minutes', '60 minutes'])
            .then((time) => {
            console.log('time:' + time);
            if (typeof time == 'undefined') {
                return false;
            }
            let timePomo = parseInt(time);
            vscode_1.window.showInformationMessage('Pomodoro start with ' + time + '!');
            let pomodoro = new Pomodoro(Status.pomodoro, timePomo * 60, shortToLongTime);
            let pomodoroController = new PomodoroController(pomodoro);
        });
    });
    context.subscriptions.push(timerStart);
}
exports.activate = activate;
var Status;
(function (Status) {
    Status["pomodoro"] = "pomodoro";
    Status["shortBreak"] = "shortBreak";
    Status["longBreak"] = "longBreak";
})(Status || (Status = {}));
class PomodoroController {
    constructor(pompdoro) {
        this._pompdoro = pompdoro;
        this._statusBarItem = vscode_1.window.createStatusBarItem(vscode_1.StatusBarAlignment.Left, 0);
        this._statusBarItem.command = 'timer.start';
        this._statusBarItem.tooltip = 'Click to start a pomodoro';
        this._statusBarItem.show();
        this._interval = setInterval(() => this.refreshUI(), 1000);
        this.refreshUI();
    }
    dispose() {
        this._statusBarItem.dispose();
        clearInterval(this._interval);
    }
    refreshUI() {
        let text = this._pompdoro.timer();
        if (text) {
            this._statusBarItem.text = text;
            this._statusBarItem.command = 'timer.cancel';
            this._statusBarItem.tooltip = 'Cancel';
        }
        else {
            this.dispose();
        }
    }
}
class Pomodoro {
    constructor(status, time, shortToLongTime) {
        this._status = status;
        this._time = time;
        this._remainTime = time;
        this._shortToLongTime = shortToLongTime - 1;
        this._breakTime = shortToLongTime - 1;
    }
    isPomodoro() {
        return Status.pomodoro == this._status();
    }
    action() {
        if (this._remainTime < 0) {
            if (this.isPomodoro()) {
                if (this._breakTime < 0) {
                    this._status = Status.shortBreak;
                    this._remainTime = 5 * 60;
                    this._breakTime--;
                }
                else if (this._breakTime == 0) {
                    this._status = Status.longBreak;
                    this._remainTime = 10 * 60;
                    this._breakTime = this._shortToLongTime;
                }
            }
            else {
                this._status = Status.pomodoro;
                this._remainTime = this._time;
            }
        }
        return this;
    }
    timer() {
        this._remainTime--;
        let text = this._remainTime + "";
        return this._status.toUpperCase() + ' in ' + text;
    }
}
//# sourceMappingURL=extension.js.map

src/extension.ts
'use strict';
// The module 'vscode' contains the VS Code extensibility API
// Import the module and reference it with the alias vscode in your code below
import {
    window,
    commands,
    workspace,
    Disposable,
    ExtensionContext,
    StatusBarAlignment,
    StatusBarItem,
    TextDocument
} from 'vscode';

// this method is called when your extension is activated
// your extension is activated the very first time the command is executed
export function activate(context: ExtensionContext) {

    // Use the console to output diagnostic information (console.log) and errors (console.error)
    // This line of code will only be executed once when your extension is activated
    console.log('Congratulations, your extension "x" is now active!');

    // Get settings
    //let settings = workspace.getConfiguration().get("tomatoTimer");
    let shortToLongTime = 3;

    let timerStart = commands.registerCommand('pomodoro.start', () => {
        window
            .showQuickPick(['10 minutes', '20 minutes', '25 minutes', '45 minutes', '60 minutes'])
            .then((time) => {
                console.log('time:' + time);
                if (typeof time == 'undefined') {
                    return false;
                }
                let timePomo = parseInt(time);
                window.showInformationMessage('Pomodoro start with ' + time + '!');
                let pomodoro = new Pomodoro(Status.pomodoro, timePomo * 60, shortToLongTime);
                let pomodoroController = new PomodoroController(pomodoro);
            });
    });

    context.subscriptions.push(timerStart);
}

enum Status {
    pomodoro = 'pomodoro',
        shortBreak = 'shortBreak',
        longBreak = 'longBreak'
}

class PomodoroController {

    private _pompdoro: Pomodoro
    private _statusBarItem;
    private _interval;

    constructor(pompdoro: Pomodoro) {
        this._pompdoro = pompdoro;
        this._statusBarItem = window.createStatusBarItem(StatusBarAlignment.Left, 0);
        this._statusBarItem.command = 'timer.start';
        this._statusBarItem.tooltip = 'Click to start a pomodoro';
        this._statusBarItem.show();

        this._interval = setInterval(() => this.refreshUI(), 1000);

        this.refreshUI();
    }

    dispose() {
        this._statusBarItem.dispose();
        clearInterval(this._interval);
    }

    refreshUI() {
        let text = this._pompdoro.timer();
        if (text) {
            this._statusBarItem.text = text;
            this._statusBarItem.command = 'timer.cancel';
            this._statusBarItem.tooltip = 'Cancel';
        } else {
            this.dispose();
        }
    }
}

class Pomodoro {

    private _status;
    private _time;
    private _remainTime;
    private _shortToLongTime;
    private _breakTime;


    constructor(status, time, shortToLongTime) {
        this._status = status;
        this._time = time;
        this._remainTime = time;
        this._shortToLongTime = shortToLongTime - 1;
        this._breakTime = shortToLongTime - 1;
    }

    isPomodoro() {
        return Status.pomodoro == this._status();
    }

    action() {
        if (this._remainTime < 0) {
            if (this.isPomodoro()) {
                if (this._breakTime < 0) {
                    this._status = Status.shortBreak;
                    this._remainTime = 5 * 60;
                    this._breakTime--;
                } else if (this._breakTime == 0) {
                    this._status = Status.longBreak;
                    this._remainTime = 10 * 60;
                    this._breakTime = this._shortToLongTime;
                }
            } else {
                this._status = Status.pomodoro;
                this._remainTime = this._time;
            }
        }
        return this;
    }

    timer() {
        this._remainTime--;
        let text = this._remainTime + ""
        return this._status.toUpperCase() + ' in ' + text;
    }
}